docs/wfe.rst

```diff
@@ -0,0 +1,216 @@
+
+::
+
+   coefficients_sequence = [0, 0, 0, 0, 35e-9]
+
+   osys = poppy.OpticalSystem()
+   circular_aperture = poppy.CircularAperture(radius=RADIUS)
+   osys.addPupil(circular_aperture)
+   thinlens = poppy.ZernikeWFE(radius=RADIUS, coefficients=coefficients_sequence)
+   osys.addPupil(thinlens)
+   osys.addDetector(pixelscale=PIXSCALE, fov_arcsec=FOV)
+
+   psf_with_zernikewfe = osys.calcPSF(wavelength=WAVELENGTH, display_intermediates=True)
+
+The resulting PSF, plotted with `poppy.display_PSF`:
+
+.. figure:: ./figures/wfe/zernikewfe_astigmatism.png
+   :scale: 50%
+
+   PSF of a 2.0 meter diameter circular aperture with oblique astigmatism applied.
+
+Using ZernikeWFE to analyze a variety of possible PSFs given a WFE budget
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The API for `ZernikeWFE` also lends itself well to generating coefficients programmatically and passing it in. Say we have an error budget where we know the following about the RMS wavefront error in the Zernike components:
+
+  * **Piston**, *j=1* — disregarded for a telescope
+  * **Tilt X**, *j=2* — ±100 nm
+  * **Tilt Y**, *j=3* — ±100 nm
+  * **Focus**, *j=4* — ±50 nm
+  * **Astigmatism 45**, *j=5* — ±36 nm
+  * **Astigmatism 0**, *j=6* — ±36 nm
+
+We can use `ZernikeWFE` to generate a library of sample PSFs satisfying this error budget. First, we write a short function that can generate coefficients from our specifications. ::
+
+   wfe_budget = [0, 100, 100, 50, 36, 36]
+
+   def generate_coefficients(wfe_budget):
+       coefficients = []
+       for term in wfe_budget:
+           coefficients.append(
+               # convert nm to meters, get value in range
+               np.random.uniform(low=-1e-9 * term, high=1e-9 * term)
+           )
+       return coefficients
+
+Now we use this to generate a few sets of coefficients. ::
+
+   possible_coefficients = [generate_coefficients(wfe_budget) for i in range(5)]
+
+Now we simply loop over the sets of coefficients, supplying them to ZernikeWFE and (optionally) plotting the results::
+
+   plt.figure(figsize=(18,2))
+
+   for idx, coefficient_set in enumerate(possible_coefficients, start=1):
+       plt.subplot(1, 5, idx)
+
+       osys = poppy.OpticalSystem()
+       hex_aperture = poppy.CircularAperture(radius=RADIUS)
+       osys.addPupil(hex_aperture)
+       thinlens = poppy.ZernikeWFE(
+           coefficients=coefficient_set,
+           radius=RADIUS
+       )
+       osys.addPupil(thinlens)
+       osys.addDetector(pixelscale=PIXSCALE, fov_arcsec=FOV)
+
+       psf = osys.calcPSF(wavelength=WAVELENGTH, display=False)
+       poppy.display_PSF(psf, title="PSF #{}".format(idx))
+
+.. figure:: ./figures/wfe/zernikewfe_wfe_budget.png
+
+   PSF of a 2.0 meter diameter circular aperture with randomized combinations of aberrations applied.
+
```

Comment:
Same comment as above - could show the OPDs as well as the PSFs. (would need to take care generate both figures from the same script to ensure consistency of the random coefficients)